### ISSUE TITLE
Load testing

### ISSUE THREAD

**Issue by lawben** [hyrise/hyrise#674]
I'm still trying to figure out why Hyrise gets stuck on certain tasks (seems like something is wrong while committing, but not sure yet) and I was thinking it might be smart to add "load" tests. 

The idea would be to have a small test bed with *n* threads that each fire *x* queries (randomly selected from query pool) to see if the system performs as it should under load. This could also be used to identify performance decrease in the Scheduler or MVCC.

Any thought on whether this is a good or bad idea or if it is worth pursuing?

**Comment by Bensk1** [comment]
„Gets stuck” means that Hyrise is not working on any new transactions?

Do you have an example for us?

**Comment by lawben** [comment]
For the Networking seminar group we tried to benchmark our server with multiple connections and some tasks are never finished, so the server cannot send a response and "freezes". I've looked into this with @mrks @mrzzzrm and @mjendruk and we haven't found the exact problem yet. This can be reproduced by simply starting *n* threads (n > 1) and sending 100 queries each. At some point, a query will not be committed. All other transactions go through, just a single one (sometimes two) doesn't finish. 

My current debug prints:
```
...
comm_id: 455 -- trans_id: 454 -- last_c_id: 455 -- next_t_id: 455
comm_id: 456 -- trans_id: 455 -- last_c_id: 456 -- next_t_id: 457
comm_id: 458 -- trans_id: 457 -- last_c_id: 458 -- next_t_id: 458
comm_id: 459 -- trans_id: 458 -- last_c_id: 459 -- next_t_id: 459
comm_id: 460 -- trans_id: 459 -- last_c_id: 460 -- next_t_id: 460
...
```

In this example Transaction 456 with commit id 457 gets "stuck", while all other transactions continue. If I understood @mjendruk correctly yesterday, this should not be able to happen.

The "stuck" transaction has the debug message:
```
"timeout: comm_id: 457 -- trans_id: 456 -- last_c_id: 469 -- next_t_id: 469"
```

This indicates that later transactions have been committed and thus, the stuck commit context will never be called  in the `TransactionManager::_try_increment_last_commit_id() --> has_next()` loop. [(code here)](https://github.com/hyrise/hyrise/blob/master/src/lib/concurrency/transaction_manager.cpp#L84-L98)

**Comment by Bensk1** [comment]
I am fine with adding some kind of integration test for multi-threading issues. The problem does not occur for n=1, no matter how many queries you send?

**Comment by lawben** [comment]
(Just for personal logging purpose:)

in `_try_increment_last_commit_id()`:
Failed to exchange 3 to 4 (contained: 4)
Failed to exchange 7 to 8 (contained: 8)
Failed to exchange 32 to 33 (contained: 33)

--> but only 33 gets stuck

**Comment by mrks** [comment]
Do you have a test case somewhere that I can run?

**Comment by lawben** [comment]
branch: lawben/server-commit and then playground.cpp

**Comment by mrks** [comment]
fixed in #973
